src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -966,6 +966,7 @@ private boolean doPrepExec(TDSWriter tdsWriter,
             if (needsPrepare 
                 && !connection.getEnablePrepareOnFirstPreparedStatementCall() 
                 && !isExecutedAtLeastOnce
+                && preparedTypeDefinitions.length() > 0
```

Comment:
Cool, thanks @TobiasSQL . That was my initial thought too. Will commit the changes.